internal/v1alpha2/controllers/bundledeployment/bundledeployment.go

```diff
@@ -97,15 +124,57 @@ func (b *bundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 // It further validates if the unpacked content conforms to the format specified in the bundledeployment.
 // If the validation is successful, it further deploys the objects on cluster. If there is an error
 // encountered in this process, an appropriate result is returned.
-func (b *bundleDeploymentReconciler) reconcile(ctx context.Context, bundleDeployment *v1alpha2.BundleDeployment) (ctrl.Result, error) {
+func (b *bundleDeploymentReconciler) reconcile(ctx context.Context, bundleDeployment *v1alpha2.BundleDeployment, bundledeploymentStore store.Store) (ctrl.Result, error) {
 
-	// Establish a filesystem view on the local system, similar to 'chroot', with the root directory
-	// determined by the bundle deployment name. All contents from various sources will be extracted and
-	// placed within this specified root directory.
-	_, err := store.NewBundleDeploymentStore(unpackpath, bundleDeployment.GetName(), afero.NewOsFs())
-	if err != nil {
+	res, err := b.unpackContents(ctx, bundleDeployment, bundledeploymentStore)
+	// result can be nil, when there is an error during unpacking. This indicates
+	// that unpacking was failed. Update the status accordingly.
+	if res == nil || err != nil {
+		setUnpackStatusFailing(&bundleDeployment.Status.Conditions, fmt.Sprintf("unpack unsuccessful %v", err), bundleDeployment.Generation)
 		return ctrl.Result{}, err
 	}
 
+	switch res.State {
+	case source.StateUnpackPending:
+		setUnpackStatusPending(&bundleDeployment.Status.Conditions, fmt.Sprintf("pending unpack"), bundleDeployment.Generation)
```

Comment:
Shouldn't we be watching pods with some unpacking label and triggering off of them changing state?